setup.sh

```diff
@@ -225,7 +231,7 @@ function _docker_image
   if ${USE_CONTAINER}
   then
     # reuse existing container specified on command line
-    ${CRI} exec "${USE_TTY}" "${CONTAINER_NAME}" "${@}"
+    ${CRI} exec "${USE_TTY}" "${CONTAINER_NAME}" "${@:+$@}"
```

Comment:
I don't understand the purpose of all these `"${@:+$@}"`.

`If $@ is not empty; then use $@`?

Reply:
@casperklein https://stackoverflow.com/questions/44567534/work-around-for-unbound-variable-in-bash-4-0-0

Reply:
Did you encounter a problem or was this added in good faith?

BASH 4.0.0 was released more than 10 years ago. And the bug got fixed in later 4.x releases (just verified it with debian9 which included BASH 4.4).

Reply:
Even with the latest bash in macOS it was throwing the error for me. It's a macOS specific fix

Reply:
Isn't Mac using Bash 3? :D

Reply:
MacOS users will install the brew bash which is 5:

```
    ~/docker-mailserver  on   issues/1979 !2  grep '_main "' setup.sh
_main "${@}"

    ~/docker-mailserver  on   issues/1979 !2  ./setup.sh
./setup.sh: line 415: @: unbound variable

    ~/docker-mailserver  on   issues/1979 !2  bash --version
GNU bash, version 5.1.8(1)-release (x86_64-apple-darwin20.3.0)
Copyright (C) 2020 Free Software Foundation, Inc.
License GPLv3+: GNU GPL version 3 or later <http://gnu.org/licenses/gpl.html>

This is free software; you are free to change and redistribute it.
There is NO WARRANTY, to the extent permitted by law.
```

Reply:
When brew installs Bash 5, I think we do not actually need this, right?

Reply:
See my code block above. Definitely needed

Reply:
Oops. I had overlooked that :)

Reply:
@NorseGaud doesn't `setup.sh` say to use `#! /bin/bash`? Does that affect the behaviour at all? Is it actually using the v5 of Bash from brew install?

Reply:
Good observation @polarathene! @NorseGaud try `bash setup.sh`

Reply:
@aendeavor @polarathene Derp... Can't believe I didn't think about that 😅.  How do we feel about using `#!/usr/bin/env bash`?

Reply:
Also, this all presumes the user has /usr/local/bin in their PATH before /bin... I personally think `${@:+$@}` is worth it.

Reply:
On my mac i needed to `hash -r` after `brew install bash` and this only works bc `/usr/local/bin` ist listed first on default environments. So if it is environment safe as long as bash4+ is installed anywhere on the system we should go with  `${@:+$@}` i think

Reply:
Not sure TBH. This construct just looks weird to me 😆 

But I guess we could make an effort to support macOS users:)

Reply:
> Isn't Mac using Bash 3 by default? 

Holy shit. YMMD 😄 Couldn't believe that's true.

Reply:
Is it worth leaving a comment with description or link about it for future maintainer reference? Or is the link to this PR with lengthy discussion thread to wade through sufficient?